File: files/2026-01-11/CSEC 11012026.py
import pandas as pd

path = "2026-01-11/Challenge 90.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=3, nrows=8, dtype={"Item": str})
test = pd.read_excel(path, usecols="F:G", skiprows=3, nrows=5, names=["Item", "Answer"], header=None, dtype={"Item": str})


result = (
    input.groupby("Item", as_index=False)
    .agg(Answer=("Date", lambda x: len(x) if len(x) > 1 else x.iloc[0]))
    .sort_values("Item")
)
print(result.equals(test))
# First answer is not correct.

File: files/2026-01-11/CSEC 11012026.R
library(tidyverse)
library(readxl)

path <- "2026-01-11/Challenge 90.xlsx"
input <- read_excel(path, range = "B4:C12", col_types = c("text", "numeric"))
test <- read_excel(path, range = "F4:G8", col_names = c("Item", "Answer"))

result = input %>%
  summarise(Answer = ifelse(n() > 1, n(), Date), .by = Item) %>%
  arrange(Item)

all.equal(result, test, check.attributes = FALSE)
# First answer is not correct.
